### ISSUE TITLE
ExecuteWorkflowOnEntity is broken when the workflow name is NOT unique

### ISSUE THREAD

**Issue by seanmcne** [seanmcne/Microsoft.Xrm.Data.PowerShell#167]
Will fix by changing the function to use the following: 

$execWFReq = New-Object Microsoft.Crm.Sdk.Messages.ExecuteWorkflowRequest
$execWFReq.EntityId = "00000000-0000-0000-0000-000000000000"
$execWFReq.WorkflowId="00000000-0000-0000-0000-000000000000" 
$conn.ExecuteCrmOrganizationRequest($execWFReq) 

Note this needs to be tested to ensure it functions but out of box workflows for call routing fail due to having a duplicate name of other system workflows.  We can either update this to use a workflowID or maybe create a new function to execute by ID, but executing by name is not technically correct as there isn't a unique constraint today :(

**Comment by seanmcne** [comment]
Updating the proposed logic/script to execute a workflow instance in Dynamics 365 (avoiding duplicates for entities that may conflict with out of the box processes:
``` powershell
$RecordId = "00000000-0000-0000-0000-000000000000"
$workflowName = "Alert After 1 hour" 
$fetch = @"
<fetch version="1.0" output-format="xml-platform" mapping="logical" distinct="false" no-lock="true">
  <entity name="workflow">
    <attribute name="workflowid" />
    <attribute name="name" />
    <filter type="and">
      <condition attribute="statecode" operator="eq" value="1" />
      <condition attribute="type" operator="eq" value="1" />
      <condition attribute="category" operator="eq" value="0" />
      <condition attribute="name" operator="eq" value="$workflowName" />
    </filter>
  </entity>
</fetch>
"@; 
$workflows = (Get-CrmRecordsByFetch -conn $conn -Fetch $fetch)
if($workflows.Count -eq 0){
   Write-Error "Workflow with name $workflowName was not found"
}elseif($workflows.Count -gt 1){
   Write-Error "Duplicate workflow found with name $workflowName"
}else{
    $execWFReq = New-Object Microsoft.Crm.Sdk.Messages.ExecuteWorkflowRequest
    $execWFReq.EntityId = $RecordId
    $execWFReq.WorkflowId=$workflows.CrmRecords.workflowid
    $conn.ExecuteCrmOrganizationRequest($execWFReq)
}
```

**Comment by seanmcne** [comment]
Adding link to the branch as a future reminder 
https://github.com/seanmcne/Microsoft.Xrm.Data.PowerShell/tree/AddExecuteWorkflowByID

**Comment by seanmcne** [comment]
Fixed in #226
